sxt/scalar25/operation/inner_product.cc

```diff
@@ -30,6 +37,25 @@
 #include "sxt/scalar25/operation/product_mapper.h"
 
 namespace sxt::s25o {
+//--------------------------------------------------------------------------------------------------
+// async_inner_product_partial
+//--------------------------------------------------------------------------------------------------
+static xena::future<s25t::element>
+async_inner_product_partial(basct::cspan<s25t::element> lhs,
+                            basct::cspan<s25t::element> rhs) noexcept {
+  auto n = lhs.size();
```

Comment:
I assume it is okay to drop the `auto n = std::min(lhs.size(), rhs.size())` here because `async_inner_product_partial` gets called on a `subspan(rng.a(), rng.size())`? That would guarantee the `lhs.size() == rhs.size()`, correct?

Reply:
Yep. And it's a static function so it's only called from async_inner_product_impl.